### ISSUE TITLE
New option ignore for selector-no-union-class-name allow specific

### ISSUE THREAD

**Issue by alexander-schranz** [kristerkari/stylelint-scss#362]
I want to use the `selector-no-union-class-name` but allow to use union-class-name for specific classes which match a pattern.

In my use case the pattern would allow using for BEM modifiers union class name but would disallow them for BEM elements.

Regex: `^--(.*)$`

The following patterns are considered warnings:

```scss
.block {
    &--modifier {

    }
}
```

The following patterns are *not* considered warnings:

```scss
.block {
    &__element {

    }
}
```

Example configuration:

```
"selector-no-union-class-name": [
    "always",
    {
        "ignore": [ "/^--(.*)$/" ]
    }
],
```

**Comment by kristerkari** [comment]
Sound good!
